kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -897,21 +899,28 @@ impl<'tcx> GotocCtx<'tcx> {
         Stmt::atomic_block(vec![skip_stmt], loc)
     }
 
-    /// An intrinsic that translates directly into `memmove`
+    /// An intrinsic that translates directly into `memmove` (for `copy`)
```

Comment:
This is a bit confusing. It will probably benefit from the refactoring I suggested since this function does more than generating code for a specific intrinsic.

Reply:
As earlier, I don't know what is wrong with this. It's not generating code for a specific intrinsic, but for a specific type of intrinsics (i.e., "copy" intrinsics). This is similar to what we already had before the audit, but using a function instead of a macro.

Reply:
The problem is that `StatementKind::CopyNonOverlapping` is not an intrinsic.